tests/testthat/test-vpc_utils.R

```diff
@@ -0,0 +1,145 @@
+# Reference link https://github.com/aws/sagemaker-python-sdk/blob/master/tests/unit/test_vpc_utils.py
+# Actual file to compare https://github.com/DyfanJones/sagemaker-r-sdk/blob/master/R/vpc_utils.R
+
+#########################################################################
+# This example is an adaptation from
+# https://github.com/aws/sagemaker-python-sdk/blob/master/tests/unit/test_vpc_utils.py
+#
+#########################################################################
+context("vpc_utils helper function")
+
+
+library(testthat)
+
+
+
+#########################################################################
+# Setup
+#########################################################################
+subnets <- c('subnet')
+security_groups <- c('sg')
+
+good_vpc_config <- list('SUBNETS_KEY' = subnets,
+                        'SECURITY_GROUP_IDS_KEY' = security_groups)
+
+foo_vpc_config <- list('SUBNETS_KEY' = subnets,
+                       'SECURITY_GROUP_IDS_KEY' = security_groups,
+                       'foo' = 1
+                       )
+#########################################################################
+
+test_that("test legacy name_from_framework_image", {
+  image_uri = "123.dkr.ecr.us-west-2.amazonaws.com/sagemaker-mxnet-py3-gpu:2.5.6-gpu-py2"
+  test_fw1 = framework_name_from_image(image_uri)
+  exp_fw1 = list("mxnet", "py3", "2.5.6-gpu-py2", NULL)
+  
+  expect_equal(test_fw1, exp_fw1)
+})
+
+
+
+
+test_that(
+  'First test', 
+  
+  {
+  
+expect_equal(NULL,NULL)
+expect_equal(subnets,NULL)
+expect_equal(NULL,security_groups)
+expect_equal(subnets,security_groups) # Needs to be updated to reflect the assert to_dict
+
+    
+  })
+
+
+test_that(
+  
+  
+  expect_equal(
+    vpc_from_list(good_vpc_config) ==list(subnets,security_groups)
+    )
+    
+  expect_equal(
+    vpc_from_list(foo_vpc_config) ==list(subnets,security_groups)
+  )
+  
+  expect_equal(
+    vpc_from_list(NULL) == list(NULL,NULL)
+  )
+  
+  expect_equal(
+    
+    vpc_from_list(NULL, do_sanitize = TRUE) == list(NULL,NULL)
+    
+    )
+  
+  
+
+
+  )
+
+
+
+# Not sure 
+
```

Comment:
testthat framework offers a function that expects errors: https://testthat.r-lib.org/reference/output-expectations.html I believe this is what you are after.

Reply:
Please refer to: https://github.com/DyfanJones/sagemaker-r-sdk/blob/master/tests/testthat/test-record_common_protobuf.R for an example